pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -464,4 +362,12 @@ class ToolDefinition:
     Note: this is currently only supported by OpenAI models.
     """
 
+    kind: ToolKind = field(default='function')
+    """The kind of tool:
+    - `'function'`: a tool that can be executed by Pydantic AI and has its result returned to the model
+    - `'output'`: a tool that passes through an output value that ends the run
+    - `'deferred'`: a tool that cannot be executed by Pydantic AI and needs to get a result from the outside.
```

Comment:
I think it's worth giving an explicit example of what "from the outside" means here, or otherwise rephrasing that. Like, saying "the tool result will be obtained from another execution, such as a frontend application" or similar.